### ISSUE TITLE
[no-parameter-properties] Support enforcing the inverse

### ISSUE THREAD

**Issue by sindresorhus** [typescript-eslint/typescript-eslint#103]
The `no-parameter-properties` rule enforces *not* using parameter properties, but I would like to enforce *use of* parameter properties whenever possible. I actually like the shorthand.

I would suggest renaming the rule to `parameter-properties` (without the `no-` prefix) and support enforcing both *use* and *non-use* of parameter properties.

---

Relevant TSLint issue: https://github.com/palantir/tslint/issues/2638

**Comment by jedmao** [comment]
I've always used them as well. What are y'all's thoughts about the argument for introducing the rule in the first place?

> Parameter properties can be confusing to those new to TypeScript as they are less explicit than other ways of declaring and initializing class members.

[`@typescript-eslint/no-parameter-properties`](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-parameter-properties.md#disallow-the-use-of-parameter-properties-in-class-constructors-no-parameter-properties)

It sounds like the argument is that it confuses people who don't know TypeScript? A bit of a weak argument.

**Comment by JoshuaKGoldberg** [comment]
Copying a relevant code snippet from https://github.com/palantir/tslint/issues/2638#issuecomment-435633998: FWIW, parameter properties are quite nice when using libraries such as Inversify:

```ts
@injectable()
class Ninja implements Warrior {

    private _katana: Weapon;
    private _shuriken: ThrowableWeapon;

    public constructor(
        @inject(TYPES.Weapon) katana: Weapon,
        @inject(TYPES.ThrowableWeapon) shuriken: ThrowableWeapon
    ) {
        this._katana = katana;
        this._shuriken = shuriken;
    }

    public fight() { return this._katana.hit(); };
    public sneak() { return this._shuriken.throw(); };
}
```

..compared to...

```ts
@injectable()
class Ninja implements Warrior {
    public constructor(
        @inject(TYPES.Weapon)
        private readonly katana: Weapon,

        @inject(TYPES.ThrowableWeapon)
        private readonly shuriken: ThrowableWeapon,
    ) {  }

    public fight() { return this.katana.hit(); };
    public sneak() { return this.shuriken.throw(); };
}
```
